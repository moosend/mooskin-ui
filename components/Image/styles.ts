import styled from 'styled-components';

// Models
import { IBoxImageComponentProps } from '../Box/model';

// Components
import { Box } from '../Box/Box';

export const StyledImage = styled(Box)<IBoxImageComponentProps>`
	display: block;
`;

StyledImage.displayName = 'StyledImage';
