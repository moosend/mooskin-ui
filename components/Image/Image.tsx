import * as React from 'react';

// Mooskin Context HoC that passes context to component props
import { withMooskinContext } from '../Styled/MooskinContextProvider';

// Models
import { IBoxImageComponentProps } from '../Box/model';

// Styled Components
import { StyledImage } from './styles';

/**
 * Image
 */
export const Image: React.FC<IBoxImageComponentProps> = withMooskinContext((props) => {
	return <StyledImage {...props} boxAs="img" />;
});

Image.defaultProps = {
	className: '',
	style: {},
};

Image.displayName = 'Image';
