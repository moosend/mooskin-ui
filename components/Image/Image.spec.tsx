import * as React from 'react';
import { Image } from './Image';

import { mount } from 'enzyme';

describe('Image', () => {
	test('renders correctly', () => {
		const tree = mount(<Image src="asd" />);
		expect(tree).toMatchSnapshot();
	});
});
