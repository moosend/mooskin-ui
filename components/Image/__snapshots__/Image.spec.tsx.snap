// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<Image
  className=""
  src="asd"
  style={Object {}}
>
  <Component
    className=""
    palette={
      Object {
        "backgroundColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "borderColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "fontColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
      }
    }
    src="asd"
    style={Object {}}
  >
    <StyledImage
      boxAs="img"
      className=""
      palette={
        Object {
          "backgroundColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "borderColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "fontColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
        }
      }
      src="asd"
      style={Object {}}
    >
      <Box
        boxAs="img"
        className="sc-gtssRu dTEizC"
        palette={
          Object {
            "backgroundColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "borderColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "fontColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
          }
        }
        src="asd"
        style={Object {}}
        transition="0.2s all"
      >
        <Component
          boxAs="img"
          className="sc-gtssRu dTEizC"
          palette={
            Object {
              "backgroundColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "borderColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "fontColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
            }
          }
          src="asd"
          style={Object {}}
          transition="0.2s all"
        >
          <StyledBox
            as="img"
            boxAs="img"
            className="sc-gtssRu dTEizC"
            palette={
              Object {
                "backgroundColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "borderColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "fontColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
              }
            }
            src="asd"
            style={Object {}}
            transition="0.2s all"
          >
            <img
              className="sc-bdnylx jknATe sc-gtssRu dTEizC"
              src="asd"
              style={Object {}}
            />
          </StyledBox>
        </Component>
      </Box>
    </StyledImage>
  </Component>
</Image>
`;
